subsys/bluetooth/audio/codec.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022 Bose Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/** @file
+ *  @brief Bluetooth LE-Audio codec LTV parsing
+ *
+ *  Helper functions to parse codec config data as specified in the Bluetooth assigned numbers for
+ *  Generic Audio.
+ */
+
+#include <bluetooth/audio/audio.h>
+#include <sys/byteorder.h>
+
+/* Overkill for now, but useful if data format is changed from bt_codec_data to TLV format. */
+void bt_codec_parse(const struct bt_codec *codec,
+		   bool (*func)(const struct bt_codec_data *data, void *user_data),
+		   void *user_data)
+{
+	for (int i = 0; i < codec->data_count; i++) {
+		if (!func(&codec->data[i], user_data)) {
+			return;
+		}
+	}
+}
+
+void bt_meta_parse(const struct bt_codec *codec,
+		   bool (*func)(const struct bt_codec_data *data, void *user_data),
+		   void *user_data)
+{
+	for (int i = 0; i < codec->meta_count; i++) {
+		if (!func(&codec->meta[i], user_data)) {
+			return;
+		}
+	}
+}
+
+bool bt_codec_get_val(const struct bt_codec *codec,
+		      uint8_t type,
+		      const struct bt_codec_data **data)
+{
+	for (int i = 0; i < codec->data_count; i++) {
+		if (codec->data[i].data.type == type) {
+			*data = &codec->data[i];
+			return true;
+		}
+	}
+	return false;
+}
+
+int bt_codec_cfg_get_freq(const struct bt_codec *codec)
+{
+	const struct bt_codec_data *element;
+
+	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_FREQ, &element)) {
```

Comment:
`element` seems like an unrelated name

```suggestion
	const struct bt_codec_data *codec_data;

	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_FREQ, &codec_data)) {
```

Reply:
I like `element` as you get one value element from the pile of LTV codec_data. Would `value` work better for you? IMO codec_data is confusing as that is also the name of the type (which is used for many different types of data)

Reply:
> I like `element` as you get one value element from the pile of LTV codec_data. Would `value` work better for you? 

> IMO codec_data is confusing as that is also the name of the type (which is used for many different types of data)

I am actually on the opposite preference: I like the format of `struct <module>_<name> <name>` e.g. `struct bt_conn *conn` and `struct bt_codec_data *codec_data` as the name of the variable often makes it clear what it covers. 

We can continue with `element`. I think `value` is confusing, as the `struct bt_codec_data` has a field called `value`, and I'm not really a fan over the `<name>-><name>` (e.g. `value->value`) format :) (even though it appears a lot for ISO atm...). 

Alternative suggestion could be `ltv` as that covers the format of the struct, which is a length-type-value structure.